modules/apps/static/portal-osgi-web/portal-osgi-web-wab-generator-impl/src/main/java/com/liferay/portal/osgi/web/wab/generator/internal/WabGenerator.java

```diff
@@ -170,29 +205,17 @@ public Void addingBundle(
 
 	@Deactivate
 	protected void deactivate(BundleContext bundleContext) throws Exception {
+		_serviceTrackerList.close();
+
 		_serviceRegistration.unregister();
 
 		_serviceRegistration = null;
 	}
 
-	@Reference(
-		cardinality = ReferenceCardinality.OPTIONAL,
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY,
-		target = "(&(original.bean=true)(bean.id=javax.servlet.ServletContext))"
-	)
-	protected void setServletContext(ServletContext servletContext) {
```

Comment:
@JanisZhang This was added in 2016 when the first OSGi Liferay was just released. Could you please double check if this still needs to be a optional, dynamic, greedy reference? Most other references using the same filter are just normal references.

If this can be a normal reference, then the `_portalIsReady` is no longer needed.